dashboard/app/models/lesson.rb

```diff
@@ -435,6 +435,44 @@ def seeding_key(seed_context)
     my_key.stringify_keys
   end
 
+  # Finds all other lessons which match the following criteria:
+  # 1. in the same curriculum umbrella (CSF/CSD/CSP) as this lesson
```

Comment:
I think this makes sense for right now but it doesn't not feel great when I start thinking about adding new kinds of courses (CSA) or allowing other people to write courses on our site. Should this instead be based on Course Offering? I guess thats not perfect because of CSF where pre-express and express contain things from course A to F.  Should we be thinking about Course Type sooner and using that here?

Reply:
Also CSF shares lessons with Hour of Code sometimes.

Reply:
Good discussion Dani. Some initial thoughts in response:
1. I thought that **Course Type**, as we had discussed it previously, has more to do with which "skin" of a course you are opting into. Just because two courses opt for a similar style, doesn't mean their content is super related.
2. **Curriculum Umbrella**, as we currently use it, is more logically similar to what we want here -- in our new terminology, I could imagine this as "group" of course offerings. That is how I am thinking about it, anyway :-) I'm not quite clear on whether we were planning to keep Curriculum Umbrella around once we had Course Offering and Course Type. I wouldn't want to keep it around just for this purpose. but if we do keep it around, then I think it is the right concept.
3. if we keep Curriculum Umbrella, then we could add CSA as a valid value for it. That doesn't solve every case, but could help a fair bit, and might be enough to last us until we start tracking explicit links between related lessons.
4. I'm not really sure how to arrange this so that CSF and HOC lessons could be identified as related to each other, or if they would even have the same lesson keys. if they don't, we might have to wait until we're explicitly storing links in the database in order to uncover those relationships.

All of this said I am open to other ideas here. my thought with this was to get something in front of curriculum writers and see what they think... but I don't want to write off your perspective here so if it seems inadequate then we don't necessarily have to wait for more feedback to improve it further.

Reply:
> 1. I thought that **Course Type**, as we had discussed it previously, has more to do with which "skin" of a course you are opting into. Just because two courses opt for a similar style, doesn't mean their content is super related.

Ya I guess that makes sense. Course Type is probably not the right choice here.

> 2. **Curriculum Umbrella**, as we currently use it, is more logically similar to what we want here -- in our new terminology, I could imagine this as "group" of course offerings. That is how I am thinking about it, anyway :-) I'm not quite clear on whether we were planning to keep Curriculum Umbrella around once we had Course Offering and Course Type. I wouldn't want to keep it around just for this purpose. but if we do keep it around, then I think it is the right concept.
> 3. if we keep Curriculum Umbrella, then we could add CSA as a valid value for it. That doesn't solve every case, but could help a fair bit, and might be enough to last us until we start tracking explicit links between related lessons.

It seems like Course Offering would do everything you want it to for CSD/CSP its just the weird way CSF works that makes us need something like curriculum umbrella. I would love to see us use Course Offering as the main way to look for this similar lessons and then maybe special case CSF only to use Curriculum Umbrella. This feels more future proof to me.

> 4. I'm not really sure how to arrange this so that CSF and HOC lessons could be identified as related to each other, or if they would even have the same lesson keys. if they don't, we might have to wait until we're explicitly storing links in the database in order to uncover those relationships.

I think this is fine to leave out for now. I don't think the lesson plans themselves are the same between in course and in HOC so probably not a big deal.

> All of this said I am open to other ideas here. my thought with this was to get something in front of curriculum writers and see what they think... but I don't want to write off your perspective here so if it seems inadequate then we don't necessarily have to wait for more feedback to improve it further.

Reply:
> It seems like Course Offering would do everything you want it to for CSD/CSP its just the weird way CSF works that makes us need something like curriculum umbrella. I would love to see us use Course Offering as the main way to look for this similar lessons and then maybe special case CSF only to use Curriculum Umbrella. This feels more future proof to me.

Great points Dani. I like the idea of special-casing CSF -- I agree that makes us more future proof, since it discourages other Curriculum Umbrellas from getting into that state.

Reply:
This is done now. I got a bit carried away with optimizing the query count but I think I found a good stopping point.